narwhal/primary/src/primary.rs

```diff
@@ -682,6 +685,15 @@ impl PrimaryReceiverHandler {
         let committee = self.committee.clone();
         header.validate(&committee, &self.worker_cache)?;
 
+        let num_parents = request.body().parents.len();
+        ensure!(
+            num_parents <= committee.size(),
+            DagError::TooManyParents(num_parents, committee.size())
+        );
```

Comment:
Do we also want to check that the parents are indeed parents referenced by the header?

Reply:
Later in `try_accept_unknown_parents()`, only requested certificates are processed. And only missing parents are requested.